app/models/author.rb

```diff
@@ -1,6 +1,9 @@
 class Author < ApplicationRecord
-  has_many :books
-  has_many :published, foreign_key: :publisher_id, class_name: 'Book', as: :publisher
+  has_many :books, dependent: :delete_all
+  has_many :published, foreign_key: :publisher_id, class_name: 'Book', as: :publisher, dependent: :delete_all, inverse_of: :author
```

Comment:
more of a note, since you added the foreign keys....the `dependent: delete_all` part is now redundant